src/core/src/op/grid_sample.cpp

```diff
@@ -27,7 +27,7 @@ bool op::v9::GridSample::visit_attributes(AttributeVisitor& visitor) {
 void op::v9::GridSample::validate_and_infer_types() {
     OV_OP_SCOPE(v9_GridSample_validate_and_infer_types);
     NODE_VALIDATION_CHECK(this,
-                          get_input_element_type(1).is_real(),
+                          get_input_element_type(1).is_real() || get_input_element_type(1).is_dynamic(),
                           "The element type of the grid input tensor must be a floating point type.");
```

Comment:
error message does not look right now

Reply:
May be it is sufficient to check: `get_input_element_type(1).is_dynamic()`? This is because it also covers real numbers.

Reply:
I think if the type is static we would like to throw if we got integral number as input. So the changes are correct I think

Reply:
I changed logic, now type check performed only if type already known (== not dynamic), so message should be valid again